### PR TITLE
Matrix Chain Multiplication in Java

### LINKED ISSUE #1784: Matrix chain multiplication in Java
@adishjain @nj4710 @ayushin78 
As a GSSoC participant, I would like to work on this, please assign this task to me.

### PR DESCRIPTION
**Pull request by eKta37**
Fixes #1784 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Adding a new file

#### Languages Used:
- Java

#### Files Added:
- Matrix_Chain_Multiplication_DP.java

### CODE REVIEW

File: Gcd/gcd.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/*
```
Comment (jainaman224) [inline Gcd/gcd.java]: Remove this file as it is not related to this issue.

File: Matrix_Chain_Multiplication/Matrix_Chain_Multiplication_DP.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
+/*
+ * A Dynamic Programming approach to find the minimum(optimal) number of operations required to multiply a chain of matrices
+ * Time complexity = O(n^3)
+ * Space complexity = O(n^2)
+ */
+import java.util.Scanner;
+
+public class matrixChainMultiplication { 
+	
+	//function to find the optimized number of operations required for multiplication using DP
```
Comment (jainaman224) [inline Matrix_Chain_Multiplication/Matrix_Chain_Multiplication_DP.java]: Use 4 space indentation.

File: Matrix_Chain_Multiplication/Matrix_Chain_Multiplication_DP.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	static int orderOfMultiplication(int arr[]) { 
+		int size = arr.length;
+		
+		//2D array to store optimal solution for all i
```
Comment (jainaman224) [inline Matrix_Chain_Multiplication/Matrix_Chain_Multiplication_DP.java]: Add space after `//`.

File: Matrix_Chain_Multiplication/Matrix_Chain_Multiplication_DP.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+		int dp[][] = new int[size][size]; 
+		
+		// cost will be zero for multiplying one matrix
+		for (int i=1; i<size; i++) {
```
Comment (jainaman224) [inline Matrix_Chain_Multiplication/Matrix_Chain_Multiplication_DP.java]: Add spaces around operators lile `=`, `==`, `+` etc.

**Comment by somya-kapoor** [comment]
@eKta37 Are you working on it?

**Comment by eKta37** [comment]
Yes, I'm working on it.

**Comment by suchi12j** [comment]
Pls assign this to me

**Comment by somya-kapoor** [comment]
@eKta37 Please use 4 space indentation

**Comment by eKta37** [comment]
@jainaman224 please review it.

**Comment by somya-kapoor** [comment]
@eKta37 Please use 4 space indentation, you are using 8 space

**Comment by somya-kapoor** [comment]
@eKta37 Are you working on it?

**Comment by somya-kapoor** [comment]
@eKta37 Due to no activity for more than 7 days, I am closing this issue.
